src/main/java/seedu/command/EditContactCommand.java

```diff
@@ -26,23 +27,25 @@ public void execute() {
         try {
             if (contactIndex == PERSONAL_CONTACT_INDEX) {
                 updatePersonalContact();
+            } else if (contactIndex == ALL_CONTACTS_INDEX) {
+                ExceptionTextUi.missingIndexEditMessage();
             } else {
                 Contact editedContact = duplicateContact(contactList.getContactAtIndex(contactIndex));
                 editedContact.editContact(contactDetails);
                 handleDuplicates(editedContact);
             }
-        } catch (IndexOutOfBoundsException | InvalidFlagException e) {
+        } catch (IndexOutOfBoundsException e) {
```

Comment:
Should InvalidFlagException still be caught here for safe measure? Considering most methods handle flags which throw InvalidFlagException, but if they are handled accordingly then there is no issue.